modules/nf-core/dysgu/main.nf

```diff
@@ -0,0 +1,56 @@
+process DYSGU {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/dysgu%3A1.6.2--py310h770aed0_0' :
+        'biocontainers/dysgu:1.6.2--py310h770aed0_0' }"
+
+    input:
+    tuple val(meta), path(input_bam), path(input_bam_index)
+    tuple val(meta2), path(fasta), path(fai)
+
+    output:
+    tuple val(meta), path('*.vcf.gz')       , emit: vcf
+    tuple val(meta), path('*.vcf.gz.tbi')   , emit: tbi
+    path 'versions.yml'                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    def args3 = task.ext.args3 ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+
+    dysgu run \\
+        -p4 \\     
+        $fasta \\
+        $input_bam \\
+        | bgzip ${args2} --threads ${task.cpus} --stdout > ${prefix}.vcf.gz
+    tabix ${args3} ${prefix}.vcf.gz
```

Comment:
```suggestion
    dysgu run \\
        -p ${task.cpus} \\     
        $fasta \\
        . \\
        $input_bam \\
        | bgzip ${args2} --threads ${task.cpus} --stdout > ${prefix}.vcf.gz
    tabix ${args3} ${prefix}.vcf.gz
```

Reply:
@famosab after running test with above modifications, there is still some errors with directory! I was thinking to  to modify `.` to `.temp` hopefully it will work?

`ValueError: Working directory already exists. Add -x / --overwrite=True to proceed, or supply --ibam to re-use temp files in working directory`  
[command.log](https://github.com/user-attachments/files/15968416/command.log)

Reply:
> @famosab after running test with above modifications, there is still some errors with directory! I was thinking to to modify `.` to `.temp` hopefully it will work?
> 
> `ValueError: Working directory already exists. Add -x / --overwrite=True to proceed, or supply --ibam to re-use temp files in working directory` [command.log](https://github.com/user-attachments/files/15968416/command.log)

> update

I have added `-x` param to overwrite the directory it works, but still assertions are failing I am trying to find the errors

Reply:
Yes probably change it to the chr21 fasta

Reply:
> Yes probably change it to the chr21 fasta

I have updated refrence_genome to `>chr21` file and still assertions are failing and there's no errors from dysgu 
only- `No events found`

Input bam

> 2024-06-25 11:36:07,112 [INFO   ]  [dysgu-run] Version: 1.6.2
> 2024-06-25 11:36:07,112 [INFO   ]  run -p 2 -x genome.fasta . test.paired_end.recalibrated.sorted.bam
> 2024-06-25 11:36:07,112 [INFO   ]  Destination: .
> 2024-06-25 11:36:07,382 [INFO   ]  dysgu fetch test.paired_end.recalibrated.sorted.bam written to ./test.paired_end.recalibrated.sorted.dysgu_reads.bam, n=252, time=0:00:00 h:m:s
> 2024-06-25 11:36:07,382 [INFO   ]  Input file is: ./test.paired_end.recalibrated.sorted.dysgu_reads.bam
> [W::hts_idx_load3] The index file is older than the data file: test.paired_end.recalibrated.sorted.bam.bai
> 2024-06-25 11:36:07,385 [INFO   ]  Sample name: normal
> 2024-06-25 11:36:07,385 [INFO   ]  Writing vcf to stdout
> 2024-06-25 11:36:07,385 [INFO   ]  Running pipeline
> 2024-06-25 11:36:07,552 [INFO   ]  Calculating insert size. Removed 0 outliers with insert size >= 1359.0
> 2024-06-25 11:36:07,563 [INFO   ]  Inferred read length 100.0, insert median 351, insert stdev 145
> 2024-06-25 11:36:07,564 [INFO   ]  Max clustering dist 1076
> 2024-06-25 11:36:07,564 [INFO   ]  Building graph with clustering 1076 bp
> 2024-06-25 11:36:07,566 [INFO   ]  Total input reads 252
> 2024-06-25 11:36:07,566 [INFO   ]  Graph constructed
> 2024-06-25 11:36:07,566 [INFO   ]  Minimum support 3
> 2024-06-25 11:36:07,602 [CRITICA]  No events found
> 2024-06-25 11:36:07,602 [INFO   ]  dysgu run test.paired_end.recalibrated.sorted.bam complete, time=0:00:00 h:m:s


Input cram
> > 2024-06-25 11:36:18,595 [INFO   ]  [dysgu-run] Version: 1.6.2
> 2024-06-25 11:36:18,595 [INFO   ]  run -p 2 -x genome.fasta . test.paired_end.recalibrated.sorted.cram
> 2024-06-25 11:36:18,595 [INFO   ]  Destination: .
> 2024-06-25 11:36:19,056 [INFO   ]  dysgu fetch test.paired_end.recalibrated.sorted.cram written to ./test.paired_end.recalibrated.sorted.dysgu_reads.bam, n=252, time=0:00:00 h:m:s
> 2024-06-25 11:36:19,057 [INFO   ]  Input file is: ./test.paired_end.recalibrated.sorted.dysgu_reads.bam
> 2024-06-25 11:36:19,058 [INFO   ]  Sample name: normal
> 2024-06-25 11:36:19,058 [INFO   ]  Writing vcf to stdout
> 2024-06-25 11:36:19,058 [INFO   ]  Running pipeline
> 2024-06-25 11:36:19,509 [INFO   ]  Calculating insert size. Removed 0 outliers with insert size >= 1359.0
> 2024-06-25 11:36:19,519 [INFO   ]  Inferred read length 100.0, insert median 351, insert stdev 145
> 2024-06-25 11:36:19,520 [INFO   ]  Max clustering dist 1076
> 2024-06-25 11:36:19,520 [INFO   ]  Building graph with clustering 1076 bp
> 2024-06-25 11:36:19,522 [INFO   ]  Total input reads 252
> 2024-06-25 11:36:19,522 [INFO   ]  Graph constructed
> 2024-06-25 11:36:19,522 [INFO   ]  Minimum support 3
> 2024-06-25 11:36:19,556 [CRITICA]  No events found
> 2024-06-25 11:36:19,556 [INFO   ]  dysgu run test.paired_end.recalibrated.sorted.cram complete, time=0:00:00 h:m:s

Reply:
seems like it does not find any varaints then,no?

Reply:
thats probably also why the files are empty, maybe it makes sense to run fydsgu on some files locally first

Reply:
> thats probably also why the files are empty, maybe it makes sense to run fydsgu on some files locally first

I've recently tested Dysgu on several BAM files from GIAB (HG002) locally, and it worked well butI will test again rn. However, it seems to be struggling to detect variants now. Could the size of the BAM files affect its detection capability?

Reply:
> > thats probably also why the files are empty, maybe it makes sense to run fydsgu on some files locally first
> 
> I've recently tested Dysgu on several BAM files from GIAB (HG002) locally, and it worked well butI will test again rn. However, it seems to be struggling to detect variants now. Could the size of the BAM files affect its detection capability?

@famosab locally dysgu is running perfectly without any errors i have reproduced results with same command line as we are using in `main.nf` file.
[result.vcf.gz](https://github.com/user-attachments/files/15971708/result.vcf.gz)
```
#!/bin/bash

# Define variables
fasta="/home/tigem/h.poddar/test_dysgu/reference/GRCh38_no_alt_analysis_set.fasta"
input_bam="/home/tigem/h.poddar/test_dysgu/input/HG003.novaseq.pcr-free.35x.dedup.grch38_no_alt.chr20.bam"
cpus=16
args2="-c"  # Define as needed
args3="-p"  # Define as needed
prefix="result"  # Define as needed

# Run Dysgu command
dysgu run \
    -p ${cpus} \
    -x \
    ${fasta} \
    . \
    ${input_bam} \
    | bgzip ${args2} --threads ${cpus} --stdout > ${prefix}.vcf.gz

# Index the gzipped VCF file using tabix
tabix ${args3} ${prefix}.vcf.gz
```
![Screenshot from 2024-06-25 15-07-26](https://github.com/nf-core/modules/assets/45700858/0f73a0c2-3c70-490a-b4dc-375755f9187d)

Reply:
Yeah the problem is not the tool itself, it might be that the test data is not suitable (because there are no detectable variants(?) ) -> can you use the test data and check locally so that we can rule out whether its just the tests data or whether there is something wrong with the command

So as input bam use the bam from the test_data repo

Reply:
> Yeah the problem is not the tool itself, it might be that the test data is not suitable (because there are no detectable variants(?) ) -> can you use the test data and check locally so that we can rule out whether its just the tests data or whether there is something wrong with the command
> 
> So as input bam use the bam from the test_data repo

I did perform some test using test_data repo and dysgu was unable to detect any SVs in given data :(. In theory I suppose dysgu isn't compatible with this test_data.
![Screenshot from 2024-06-25 15-41-18](https://github.com/nf-core/modules/assets/45700858/9cb98d8b-b320-4ae7-9b8d-a838514b8331)

Reply:
did you also check data used for other sv callers (tiddit, manta...)?

Reply:
> did you also check data used for other sv callers (tiddit, manta...)?

Yes I did the data is same as the above for cram and bam files, Perhaps there are no SVs present in the files. Alternatively I am trying adjusting the min-support parameter, it is set at 3 by default for paired-end reads.